src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+    return trimTrailingSlashes(this.runtimeConfig.rootFilepath);
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   *
+   * @returns Absolute path of the file.
+   */
+  public mapUrlToFilePath(identifier: ResourceIdentifier, id = ''): string {
+    return this.getAbsolutePath(this.getRelativePath(identifier), id);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The file path.
+   *
+   * @throws {@Link Error}
+   * If the file path does not match the rootFilepath path of the store.
+   *
+   * @returns Url of the file.
+   */
+  public mapFilePathToUrl(path: string): string {
+    if (!path.startsWith(this.rootFilepath)) {
+      throw new Error(`File ${path} is not part of the file storage at ${this.rootFilepath}.`);
+    }
+    return this.baseRequestURI + path.slice(this.rootFilepath.length);
+  }
+
+  /**
+   * Get the content type from a file path, using its extension.
+   * @param path - The file path.
+   *
+   * @returns Content type of the file.
+   */
+  public getContentTypeFromExtension(path: string): string {
+    const extension = /\.([^./]+)$/u.exec(path);
+    return (extension && this.types[extension[1].toLowerCase()]) || APPLICATION_OCTET_STREAM;
+  }
+
+  /**
+   * Get the absolute file path based on the rootFilepath of the store.
+   * @param path - The relative file path.
+   * @param identifier - Optional identifier to add to the path.
+   *
+   * @returns Absolute path of the file.
+   */
+  public getAbsolutePath(path: string, identifier = ''): string {
+    return joinPath(this.rootFilepath, path, identifier);
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   *
+   * @returns A string representing the relative path.
+   */
+  public getRelativePath(identifier: ResourceIdentifier): string {
+    if (!identifier.path.startsWith(this.baseRequestURI)) {
+      throw new NotFoundHttpError();
+    }
+    return identifier.path.slice(this.baseRequestURI.length);
+  }
+
+  /**
+   * Splits the identifier into the parent directory and its own file name.
+   * If the identifier specifies a directory, slug will be undefined.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link ConflictHttpError}
+   * If the root identifier is passed.
+   *
+   * @returns A PathAbstraction object containing path and (optional) slug fields.
+   */
+  public parseIdentifier(identifier: ResourceIdentifier): PathAbstraction {
```

Comment:
To re-iterate the comment from Ruben: "Also, parse into what? (for naming the function)". Yes parseIdentifier was the original name but now that we're changing everything might as well give it a name that tells us what this function is going to do for us. E.g. `extractSlug` (why did it have to be called slug...).

Reply:
Yes I was quite confused on how to name it. I took the naming of how it was in file store for [ , path, slug] but it could also just be [ , directory, file ] something like that, with file being optional.

Reply:
My last comment was mostly about the fact that `extractSlug` doesn't sound like a fun thing to do in real life.

Reply:
Hahah luckily we're programming. I will change it to that then :)